### ISSUE TITLE
The wheels are shaking and the task fails to complete.

### ISSUE THREAD

**Issue by xss1006** [Michal-Kolomanski/Autonomous-driving-in-Carla#10]
Thank you very much for your code, but I found a place that bothered me during the running.
When I run the code for AtoB, whether using a pre-trained model or retraining it, the car just keeps spinning its wheels for a long time without moving forward, and it fails to complete the route. What could be the reason for this?

**Comment by Michal-Kolomanski** [comment]
> Thank you very much for your code, but I found a place that bothered me during the running. When I run the code for AtoB, whether using a pre-trained model or retraining it, the car just keeps spinning its wheels for a long time without moving forward, and it fails to complete the route. What could be the reason for this?

Thanks for reaching out! Can you elaborate a little bit more about the details? Which track did you use? Can you provide some screenshots? The same happens when you train the model from scratch or use an already pre-trained model?

**Comment by xss1006** [comment]
> > 非常感谢您的代码，但我在运行过程中发现了一个困扰我的地方。当我运行 AtoB 的代码时，无论是使用预训练模型还是重新训练它，汽车都会长时间保持车轮旋转而不会向前移动，并且无法完成路线。这可能是什么原因造成的？
> 
> 感谢您的联系！您能详细说明一下吗？您使用了哪条赛道？您能提供一些截图吗？当您从头开始训练模型或使用已经预先训练过的模型时，也会发生同样的情况？

**Comment by xss1006** [comment]
> > Thank you very much for your code, but I found a place that bothered me during the running. When I run the code for AtoB, whether using a pre-trained model or retraining it, the car just keeps spinning its wheels for a long time without moving forward, and it fails to complete the route. What could be the reason for this?
> 
> Thanks for reaching out! Can you elaborate a little bit more about the details? Which track did you use? Can you provide some screenshots? The same happens when you train the model from scratch or use an already pre-trained model?

Thanks for your reply, I'll record a demo for you

[Uploading 5367b825d764121794fbf45984104c6f.zip…]()

**Comment by xss1006** [comment]
> > 非常感谢您的代码，但我在运行过程中发现了一个困扰我的地方。当我运行 AtoB 的代码时，无论是使用预训练模型还是重新训练它，汽车都会长时间保持车轮旋转而不会向前移动，并且无法完成路线。这可能是什么原因造成的？
> 
> 感谢您的联系！您能详细说明一下吗？您使用了哪条赛道？您能提供一些截图吗？当您从头开始训练模型或使用已经预先训练过的模型时，也会发生同样的情况？

Yes, this happens, especially if you train from scratch, it will be better to use a pre-trained model in the later stage, but there are also cases where the wheels are shaking

**Comment by xss1006** [comment]
> > Thank you very much for your code, but I found a place that bothered me during the running. When I run the code for AtoB, whether using a pre-trained model or retraining it, the car just keeps spinning its wheels for a long time without moving forward, and it fails to complete the route. What could be the reason for this?
> 
> Thanks for reaching out! Can you elaborate a little bit more about the details? Which track did you use? Can you provide some screenshots? The same happens when you train the model from scratch or use an already pre-trained model?

Thanks for your reply, I'll record a demo for you

https://github.com/user-attachments/assets/583fd4ac-c05d-433a-9cb8-02a4a697a2e2

**Comment by xss1006** [comment]
> > 非常感谢您的代码，但我在运行过程中发现了一个困扰我的地方。当我运行 AtoB 的代码时，无论是使用预训练模型还是重新训练它，汽车都会长时间保持车轮旋转而不会向前移动，并且无法完成路线。这可能是什么原因造成的？
> 
> 感谢您的联系！您能详细说明一下吗？您使用了哪条赛道？您能提供一些截图吗？当您从头开始训练模型或使用已经预先训练过的模型时，也会发生同样的情况？

**Comment by xss1006** [comment]
> > 非常感谢您的代码，但我在运行过程中发现了一个困扰我的地方。当我运行 AtoB 的代码时，无论是使用预训练模型还是重新训练它，都会自动长时间保持旋转而不向向前移动，并且无法完成路线。这可能是什么原因造成的？
> 
> 感谢您的联系！您能详细说明一下吗？您使用了哪条试验？您能提供一些截图吗？当您从头开始训练模型或使用已经预先训练过的模型时，也会发生同样的情况？

This is the code recorded during the pre-training process, and you can see that the wheels are shaking but not moving forward, which is more noticeable in the code trained from scratch and does not run the entire distance

https://github.com/user-attachments/assets/0ea0b2d8-5f96-44f1-b072-e57bb9db7c13

**Comment by Michal-Kolomanski** [comment]
> > > 非常感谢您的代码，但我在运行过程中发现了一个困扰我的地方。当我运行 AtoB 的代码时，无论是使用预训练模型还是重新训练它，都会自动长时间保持旋转而不向向前移动，并且无法完成路线。这可能是什么原因造成的？
> > 
> > 
> > 感谢您的联系！您能详细说明一下吗？您使用了哪条试验？您能提供一些截图吗？当您从头开始训练模型或使用已经预先训练过的模型时，也会发生同样的情况？
> 
> This is the code recorded during the pre-training process, and you can see that the wheels are shaking but not moving forward, which is more noticeable in the code trained from scratch and does not run the entire distance
> 
>  sunlogin_20240717211558.mp4

I see that you use one of the latest versions of Carla. 
- Route lines are completely different: the model was trained on other visuals. Please see the gifs from the Readme.
- Also it seems that reward, black points across the route line are not visiable on this version of Carla/graphic
- Try to lower the graphic quality as much as possible. I used the lowest graphic possible. 
- The model was trained on the Carla 0.9.10 version - please try this version. It should work as intended

**Comment by xss1006** [comment]
> > > > 非常感谢您的代码，但我在运行过程中发现了一个困扰我的地方。当我运行 AtoB 的代码时，无论是使用预训练模型还是重新训练它，都会自动长时间保持旋转而不向前方移动，并且无法完成路线。这可能是什么原因造成的？
> > > 
> > > 
> > > 感谢您的联系！您能详细说明一下吗？您使用了哪条试验？您能提供一些截图吗？当您从头开始训练模型或使用已经预先训练过的模型时，也会发生同样的情况？
> > 
> > 
> > 这是预训练过程中记录的代码，可以看到车轮在抖动但没有向前移动，这在从零开始训练且没有跑完全程的代码中更为明显
> > sunlogin_20240717211558.mp4
> 
> 我发现您使用的是最新版本的 Carla 之一。
> 
> * 路线完全不同：该模型是在其他视觉效果上训练的。请参阅自述文件中的 gif。
> * 此外，似乎奖励，路线上的黑点在这个版本的卡拉/图形上是不可见的
> * 尝试尽可能降低图形质量。我使用了最低的图形质量。
> * 该模型是在 Carla 0.9.10 版本上训练的 - 请尝试此版本。它应该可以按预期工作

Hello, thank you very much for your answer. Regarding what you said, I have made an attempt
1：I noticed there is code for randomly selecting a scene for training. Can i know how you choose which scenes to train on?
2：Using sensor images as input to the model in the code, will reducing image quality affect the training results?

**Comment by xss1006** [comment]
> > > > 非常感谢您的代码，但我在运行过程中发现了一个困扰我的地方。当我运行 AtoB 的代码时，无论是使用预训练模型还是重新训练它，都会自动长时间保持旋转而不向向前移动，并且无法完成路线。这可能是什么原因造成的？
> > > 
> > > 
> > > 感谢您的联系！您能详细说明一下吗？您使用了哪条试验？您能提供一些截图吗？当您从头开始训练模型或使用已经预先训练过的模型时，也会发生同样的情况？
> > 
> > 
> > This is the code recorded during the pre-training process, and you can see that the wheels are shaking but not moving forward, which is more noticeable in the code trained from scratch and does not run the entire distance
> > sunlogin_20240717211558.mp4
> 
> I see that you use one of the latest versions of Carla.
> 
> * Route lines are completely different: the model was trained on other visuals. Please see the gifs from the Readme.
> * Also it seems that reward, black points across the route line are not visiable on this version of Carla/graphic
> * Try to lower the graphic quality as much as possible. I used the lowest graphic possible.
> * The model was trained on the Carla 0.9.10 version - please try this version. It should work as intended

Hello, thank you very much for your answer. Regarding what you said, I have made an attempt
1：I noticed there is code for randomly selecting a scene for training. I'm training using scenario7. Do I need to randomly select routes for training？
2：Using sensor images as input to the model in the code, will reducing image quality affect the training results?

**Comment by Michal-Kolomanski** [comment]
> > > > > 非常感谢您的代码，但我在运行过程中发现了一个困扰我的地方。当我运行 AtoB 的代码时，无论是使用预训练模型还是重新训练它，都会自动长时间保持旋转而不向向前移动，并且无法完成路线。这可能是什么原因造成的？
> > > > 
> > > > 
> > > > 感谢您的联系！您能详细说明一下吗？您使用了哪条试验？您能提供一些截图吗？当您从头开始训练模型或使用已经预先训练过的模型时，也会发生同样的情况？
> > > 
> > > 
> > > This is the code recorded during the pre-training process, and you can see that the wheels are shaking but not moving forward, which is more noticeable in the code trained from scratch and does not run the entire distance
> > > sunlogin_20240717211558.mp4
> > 
> > 
> > I see that you use one of the latest versions of Carla.
> > 
> > * Route lines are completely different: the model was trained on other visuals. Please see the gifs from the Readme.
> > * Also it seems that reward, black points across the route line are not visiable on this version of Carla/graphic
> > * Try to lower the graphic quality as much as possible. I used the lowest graphic possible.
> > * The model was trained on the Carla 0.9.10 version - please try this version. It should work as intended
> 
> Hello, thank you very much for your answer. Regarding what you said, I have made an attempt 1：I noticed there is code for randomly selecting a scene for training. I'm training using scenario7. Do I need to randomly select routes for training？ 2：Using sensor images as input to the model in the code, will reducing image quality affect the training results?

1) You dont need to randomly select routes for training. You can pick just one, but then the model will potentially more overfit to that specific scenario, making it less generic. 
2) The model uses pictures as an input for the training. The quality of graphics makes a difference. Just pick one level of graphic and stick to it. However, make sure that reward points are also visible across the route line. My model was trained on the lowest graphic possible, so it wont perform well in high graphic standards.
